src/lib/cfa/ntp_conf.rb

```diff
@@ -239,8 +243,22 @@ def self.new_from_augeas(augeas_entry)
       # @param [string] key
       def self.record_class(key)
         entry_type = key.gsub("[]", "")
-        record_class = "::CFA::NtpConf::#{entry_type.capitalize}Record"
-        Kernel.const_get(record_class)
+        record_class = "#{entry_type.capitalize}Record"
+        if CFA::NtpConf.constants.include?(record_class.to_sym)
+          CFA::NtpConf.const_get(record_class.to_sym)
+        else
+          # it is not predefined record type, so use generic misc record
```

Comment:
NP: what about *it is not `a` predefined record type, so `a` generic misc record is used*